money-api/src/main/java/com/comcast/money/api/IdGenerator.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2012 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.api;
+
+import java.util.Locale;
+import java.util.Random;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class IdGenerator {
+    private IdGenerator() { }
+
+    public static final String INVALID_TRACE_ID = "00000000-0000-0000-0000-000000000000";
+    public static final String INVALID_TRACE_ID_HEX = "00000000000000000000000000000000";
+    public static final long INVALID_ID = 0L;
+    public static final String INVALID_ID_HEX = "0000000000000000";
+
+    private static final Pattern TRACE_ID_PATTERN = Pattern.compile("^([0-9a-f]{8})-?([0-9a-f]{4})-([0-9a-f]{4})-?([0-9a-f]{4})-?([0-9a-f]{12})$", Pattern.CASE_INSENSITIVE);
+    private static final Pattern TRACE_ID_HEX_PATTERN = Pattern.compile("^(?:[0-9a-f]{16}){1,2}$", Pattern.CASE_INSENSITIVE);
+    private static final Pattern ID_HEX_PATTERN = Pattern.compile("^[0-9a-f]{16}$", Pattern.CASE_INSENSITIVE);
+
+    private static final ThreadLocal<Random> threadLocalRandom = ThreadLocal.withInitial(Random::new);
+
+    /**
+     * Generates a random trace ID.
+     */
+    public static String generateRandomTraceId() {
+        long hi, lo;
+        Random random = threadLocalRandom.get();
+        do {
+            hi = random.nextLong();
+            lo = random.nextLong();
```

Comment:
do we know how random this is?

Reply:
Does it help that this is the implementation that the OTel SDK uses?

https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java#L31

I did notice that `ThreadLocalRandom` is actually a class provided by Java 8, I can at least switch the impl to using that.

Reply:
Yeah I would be okay with defaulting to what otel does here. I won't have time to dig in to this tonight. But I have set aside some time early tomorrow morning to double check